scanpipe/pipes/purldb.py

```diff
@@ -77,7 +77,9 @@ def request_get(url, payload=None, timeout=None):
     if not url:
         return
 
-    params = {"format": "json"}
+    params = {}
+    if "format=json" not in url:
+        params.update({"format": "json"})
```

Comment:
Before I created the POST-based API endpoint for Package filtering on purldb, I used the query params-based filtering on the `api/packages` endpoint. I would use the `next` link from the response to visit the next page for more package data. The `next` link already has `format=json` appended to it. When I would use the `next` link from the response to collect the package data, `format=json` would continuously be appended to the url. This caused issues when visiting multiple pages of results. I forget what error was returned by the server, but it was due to the number of query parameters I was passing.

Reply:
Thanks for the explanation, but we need to find a better way than depending on a string that may be in the URL.

What if we were to set the format as the content-type header instead of passing this as an argument? We should apply this header upfront for the whole session and do not worry about the format in the rest of the code.
```
session = requests.Session()
session.headers.update({"Content-Type": "application/json"})
```
Could you check if this would work with your code? (and the issue described above)

Reply:
@tdruez 

When I added `session.headers.update({"Content-Type": "application/json"})` to `scanpipe/pipes/purldb.py`, the POST requests made to purldb would not work. I get the message `WARNING Bad Request: /api/packages/filter_by_checksums/` on purldb.

Reply:
Ok, let's keep the exception for now and we'll deal with side effects if any.